### PR TITLE
Include `aptos-typescript-sdk-origin-method` header on server request

### PR DESCRIPTION
**Pull request by 0xmaayan**
### Description
Include the method used in the sdk when making a server call in the request header to analyze user usage

### CODE REVIEW

File: tests/e2e/client/aptosRequest.test.ts [hunk 1]
```diff
@@ -58,6 +58,10 @@ describe("aptos request", () => {
           expect(response.config.headers).toHaveProperty("x-aptos-client", `aptos-typescript-sdk/${VERSION}`);
           expect(response.config.headers).toHaveProperty("my", "header");
           expect(response.config.headers).toHaveProperty("content-type", "application/x.aptos.signed_transaction+bcs");
+          expect(response.config.headers).toHaveProperty(
+            "aptos-typescript-sdk-origin-method",
+            "test request includes all headers",
+          );
```
Comment (gregnazario) [inline tests/e2e/client/aptosRequest.test.ts:64]: Generally, we would start the headers with `x-aptos...`

**Review by gregnazario — COMMENTED** [review]
Do we know if this will get rejected from the devnet / mainnet API?  We've had that happen in the past when we added headers that weren't expected.

**Comment by 0xmaayan** [comment]
> Do we know if this will get rejected from the devnet / mainnet API? We've had that happen in the past when we added headers that weren't expected.

we dont have this restriction anymore

File: src/client/core.ts [hunk 2]
```diff
@@ -26,11 +26,12 @@ const errors: Record<number, string> = {
  * returns the response.
  */
 export async function request<Req, Res>(options: ClientRequest<Req>, client: Client): Promise<ClientResponse<Res>> {
-  const { url, method, body, contentType, params, overrides } = options;
+  const { url, method, body, contentType, params, overrides, originMethod } = options;
   const headers: Record<string, string | AnyNumber | boolean | undefined> = {
     ...overrides?.HEADERS,
     "x-aptos-client": `aptos-typescript-sdk/${VERSION}`,
     "content-type": contentType ?? MimeType.JSON,
+    "aptos-typescript-sdk-origin-method": originMethod,
```
[the remark this answers is not in the document]
Comment (0xmaayan) [reply src/client/core.ts:34]: If it is undefined, the header would not be included

**Pull request merged**